backend/src/main/java/com/dreamcatcher/mobile/controller/UserController.java

```diff
@@ -0,0 +1,54 @@
+package com.dreamcatcher.mobile.controller;
+
+import com.dreamcatcher.mobile.dto.UserAccountCreationDTO;
+import com.dreamcatcher.mobile.dto.UserAccountDTO;
+import com.dreamcatcher.mobile.entity.User;
+import com.dreamcatcher.mobile.service.UserManagementService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+public class UserController {
+
+    private UserManagementService userManagementService;
+
+    @Autowired
```

Comment:
Even better:
you can use `@RequiredArgsConstructor` or `@AllArgsConstructor` that will automatically inject the dependencies such as the `UserManagementService`. First only injects fields that are `private final` and the other injects both final and non-final fields.

The best practice is to use the `@RequiredArgsConstructor` with final fields whenever possible to avoid leaving the possibility of overriding these fields during the runtime.